File: bungee/src/main/java/com/emanuelvini/nuven/core/bungee/registry/PacketListenerRegistry.java
package com.emanuelvini.nuven.core.bungee.registry;

import com.emanuelvini.nuven.core.bungee.BungeeMain;
import com.emanuelvini.nuven.core.bungee.packet.ProfilePacketListener;
import lombok.AllArgsConstructor;

@AllArgsConstructor
public class PacketListenerRegistry {

    private BungeeMain plugin;

    public void register() {
        plugin.getProxy().getPluginManager().registerListener(plugin, new ProfilePacketListener(plugin.getProfileRepository(), plugin));
    }

}

File: bungee/src/main/kotlin/com/emanuelvini/nuven/core/bungee/registry/registries/PacketListenerRegistry.kt
package com.emanuelvini.nuven.core.bungee.registry.registries

import com.emanuelvini.nuven.core.bungee.BungeeMain
import com.emanuelvini.nuven.core.bungee.listener.packet.ProfilePacketListener
import com.emanuelvini.nuven.core.bungee.registry.BungeeRegistry

class PacketListenerRegistry(plugin: BungeeMain) : BungeeRegistry(plugin) {
    override fun register() {
        plugin.bungeeLogger.success("Conectado a database com sucesso.")
        plugin.proxy.registerChannel("nvcore:main")
        plugin.proxy.pluginManager
            .registerListener(plugin, ProfilePacketListener(plugin.proxy, plugin.proxy.scheduler, plugin.profileRepository!!, plugin ))
        plugin.bungeeLogger.success("Canais de comunicações registrados com sucesso.")
    }
}
